### PR TITLE
Include Authenticable concern in InternalApi::V1::ApplicationController

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
Closes
https://www.notion.so/saeloun/Redevelop-Sign-In-page-using-React-1ccc0724feb24e7485a02bdf4c51a17a
https://www.notion.so/saeloun/Redevelop-Sign-Up-page-using-React-58ea46ef7bbd4b239b191ca1863d1063

Why
Currently, we have authentication views on rails views. We want to move our routing completely to react and use rails API to handle requests.

What
This PR is part of the feature:move-authentication to react. In this PR I did

- Include `Authenticable` concern in `InternalApi::V1::ApplicationController`
- `skip_before_action :authenticate_user_using_x_auth_token` from `InternalApi::V1::WiseController`
- Fixed all the failing specs

### CODE REVIEW

**Comment by vipulnsward** [comment]
@apoorv1316 please run a smoke test on this

**Comment by apoorv1316** [comment]
@prasanthchaduvula Please fix the error while creating a new company.

File: app/controllers/internal_api/v1/timezones_controller.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class InternalApi::V1::TimezonesController < InternalApi::V1::ApplicationController
+  skip_before_action :authenticate_user_using_x_auth_token
```
Comment (prasanthchaduvula) [inline app/controllers/internal_api/v1/timezones_controller.rb:4]: Will remove it once we move to react routing

**Comment by prasanthchaduvula** [comment]
@apoorv1316 Please run the smoke test again

File: app/controllers/internal_api/v1/companies_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class InternalApi::V1::CompaniesController < InternalApi::V1::ApplicationController
+  skip_before_action :authenticate_user_using_x_auth_token, only: :create
```
Comment (prasanthchaduvula) [inline app/controllers/internal_api/v1/companies_controller.rb:4]: Will remove it once we move to react routing

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**